client/editorQueryEvaluator.ts

```diff
@@ -0,0 +1,221 @@
+'use strict';
+import { ExtensionContext, TextDocument, TextEdit, TextEditor, Uri, WorkspaceEdit,
+    commands, window, workspace, WorkspaceConfiguration } from 'vscode';
+import { cascadeOverrideClient } from './vscQueryParameterTools';
+import { ClientContext, sendJSQuery, sendSparql, sendXQuery, sendRows } from './marklogicClient';
+import { ClientResponseProvider } from './clientResponseProvider';
+import { contentType, RowsResponse } from 'marklogic';
+import * as ml from 'marklogic';
+
+export enum EditorQueryType {
+    JS,
+    XQY,
+    SPARQL,
+    SQL,
+    ROWS
+}
+
+export class EditorQueryEvaluator {
+    private static FOPTIONS = { tabSize: 2, insertSpaces: true };
+    private static FCOMMAND = 'vscode.executeFormatDocumentProvider';
+    private static SJS = 'sjs';
+    private static XQY = 'xqy';
+
+    private context: ExtensionContext;
+    private provider: ClientResponseProvider;
+
+    public constructor(context: ExtensionContext, provider: ClientResponseProvider) {
+        this.context = context;
+        this.provider = provider;
+    }
+
+    // Exported only for testing purposes
+    public static getQueryTextFromEditor(editor: TextEditor): string {
+        const selection = editor.selection;
+        const selectionText = editor.document.getText(selection);
+        let actualQuery: string = null;
+        if (selectionText) {
+            actualQuery = selectionText;
+        } else {
+            actualQuery = editor.document.getText();
+        }
+        return actualQuery;
+    }
+
+    public editorQuery(
+        editorQueryType: EditorQueryType, editor: TextEditor, rowsResponseFormat?: ml.RowsResponseFormat
+    ): void {
+        const actualQuery = EditorQueryEvaluator.getQueryTextFromEditor(editor);
```

Comment:
I think this line shows why `query` is sufficient - it's using two terms for the same thing here = "query text" and "actual query". I think this is sufficient and simple:

    const query = EditorQueryEvaluator.getQueryFromEditor(editor);